src/fields/date/date.ts

```diff
@@ -0,0 +1,57 @@
+import { DateTimeFormatter, LocalDate, ResolverStyle } from "@js-joda/core";
+import { Locale } from "@js-joda/locale_en-us";
+import * as Yup from "yup";
+import { ERROR_MESSAGES } from "../../shared";
+import { IFieldGenerator } from "../types";
+import { IDateSchema } from "./types";
+
+const isValidDate = (value: string, formatter: DateTimeFormatter): boolean => {
+	if (!value || value === ERROR_MESSAGES.DATE.INVALID) return false;
+	try {
+		LocalDate.parse(value, formatter);
+		return true;
+	} catch (error) {
+		return false;
+	}
+};
+
+export const date: IFieldGenerator<IDateSchema> = (id, { dateFormat = "uuuu-MM-dd", validation }) => {
+	const dateFormatter = DateTimeFormatter.ofPattern(dateFormat)
+		.withResolverStyle(ResolverStyle.STRICT)
+		.withLocale(Locale.ENGLISH);
+	const futureRule = validation?.find((rule) => "future" in rule);
+	const pastRule = validation?.find((rule) => "past" in rule);
+	const notFutureRule = validation?.find((rule) => "notFuture" in rule);
+	const notPastRule = validation?.find((rule) => "notPast" in rule);
+
+	return {
+		[id]: {
+			yupSchema: Yup.string()
+				.test("is-date", "Invalid date", (value) => {
```

Comment:
should this be using the shared error message constant?

```suggestion
				.test("is-date", ERROR_MESSAGES.DATE.INVALID, (value) => {
```